onnxruntime/core/providers/tvm/hash_alg/sha256.cc

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Implementation of SHA-256 hash algorithm was gotten from https://www.programmingalgorithms.com/algorithm/sha256/cpp/
```

Comment:
What were the other options? I'm wondering what other libraries like torch chose?

Reply:
Yes, there are other options. When I needed to use sha256 hash algorithm I thought that I will find it in standard library, but there is no. My further search showed that there are several open source libs with set of hash algorithms, `hash` lib from python and hash lib in openssl. I do not want use python API, in our case c++ API is preferable. Installation of the 3d-parties library for the sake of one algorithm look not reasonable for me. Therefore I simply copied sha256 algorithm. After your comment I rechecked options and found the same, I did not find hash lib related to torch. I agree that it will be better to use high-level API for sha256 with hidden implementation inside 3rd-parties library of ONNX Runtime. If you know such lib, API, please give me know, unfortunately I did not find it.

Reply:
Hello @xadupre! I've decomposed this PR on two parts: 1. support of TVMso EP and 2. handshake machanism. For the latter the separated PR was opened (#11437 ). Looks like that current licence for sha256 is OK, but it is important question and we will discuss it once more. As result neccessary updates will be done for other PR. Could you restart CI tests for this PR?